### PR TITLE
Use only server message body in iOS and format conversation names better.

### PR DESCRIPTION
**Pull request by MarcoPolo**
@joshblum @keybase/react-hackers

### CODE REVIEW

File: go/chat/utils/utils.go [hunk 1]
```diff
@@ -2383,6 +2383,23 @@ func GetUnverifiedConv(ctx context.Context, g *globals.Context, uid gregor1.UID,
 	return inbox.ConvsUnverified[0], nil
 }
 
+func FormatConversationName(info chat1.ConversationInfoLocal, myUsername string) string {
+	if info.TopicName != "" {
+		return fmt.Sprintf("%s#%s", info.TlfName, info.TopicName)
+	}
+	users := strings.Split(info.TlfName, ",")
+	if len(users) > 1 {
+		usersWithoutYou := make([]string, 0, len(users)-1)
```
Comment (joshblum) [inline go/chat/utils/utils.go]:
```suggestion
		usersWithoutYou := []string{}
```

Comment (MarcoPolo) [reply go/chat/utils/utils.go]: ? That wouldn't prealloc the array (not that it matters for performance of course)

Comment (joshblum) [reply go/chat/utils/utils.go]: yeah it's just more readable.

File: shared/ios/Keybase/Pusher.m [hunk 2]
```diff
@@ -33,8 +33,12 @@ - (void)localNotification:(NSString*)ident msg:(NSString*)msg badgeCount:(long)b
 
 - (void)displayChatNotification:(KeybaseChatNotification *)notification {
   NSString* ident = [NSString stringWithFormat:@"%@:%ld", notification.convID, notification.message.id_];
-  NSString* msgTxt = notification.isPlaintext ? notification.message.plaintext : @"Encrypted Message...";
-  NSString* msg = [NSString stringWithFormat:@"%@ (%@): %@", notification.message.from.keybaseUsername, notification.conversationName, msgTxt];
+  NSString* msg;
+  if (notification.isPlaintext && [notification.message.plaintext length] != 0) {
+    msg = [NSString stringWithFormat:@"%@ (%@): %@", notification.message.from.keybaseUsername, notification.conversationName, notification.message.plaintext];
```
Comment (joshblum) [inline shared/ios/Keybase/Pusher.m]: what do you think of this instead:
```
   NSString* ident = [NSString stringWithFormat:@"%@:%ld", notification.convID, notification.message.id_];
   NSString* msg;
   if (notification.isPlaintext && [notification.message.plaintext length] != 0) {
-    msg = [NSString stringWithFormat:@"%@ (%@): %@", notification.message.from.keybaseUsername, notification.conversationName, notification.message.plaintext];
+      if([ notification.message.from.keybaseUsername isEqualToString: notification.conversationName ]) {
+        msg = [NSString stringWithFormat:@"%@ sent you a message: %@", notification.message.from.keybaseUsername, notification.message.plaintext];
+      } else {
+        msg = [NSString stringWithFormat:@"%@ sent you a message in %@: %@", notification.message.from.keybaseUsername, notification.conversationName, notification.message.plaintext];
+      }
   } else {
     msg = notification.message.serverMessage;
   }
```

Comment (MarcoPolo) [reply shared/ios/Keybase/Pusher.m]: That plaintext non 1-1 case is too verbose no?

Comment (joshblum) [reply shared/ios/Keybase/Pusher.m]: i preferred it but up to you, we can always tweak these. i think it's a slightly better UX with the verbose verison

**Pull request merged**